packages/security_center/lib/services/disk_encryption_service.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'disk_encryption_service.freezed.dart';
+part 'disk_encryption_service.g.dart';
+
+/// Dialog state for managing the recovery key check process.
+@freezed
+sealed class CheckRecoveryKeyDialogState with _$CheckRecoveryKeyDialogState {
+  factory CheckRecoveryKeyDialogState.empty() =
+      CheckRecoveryKeyDialogStateEmpty;
+  factory CheckRecoveryKeyDialogState.input(String keyToCheck) =
+      CheckRecoveryKeyDialogStateInput;
+  factory CheckRecoveryKeyDialogState.result(bool valid) =
+      CheckRecoveryKeyDialogStateResult;
+  factory CheckRecoveryKeyDialogState.loading() =
+      CheckRecoveryKeyDialogStateLoading;
+  factory CheckRecoveryKeyDialogState.error(Exception e) =
+      CheckRecoveryKeyDialogStateError;
+}
```

Comment:
a final nitpick: could you move this to the `disk_encryption_providers.dart`, as it's unrelated to the service